src/utils/codemods/findAvailableCodemodsInNodeModules.js

```diff
@@ -2,32 +2,103 @@ const log = require('@dhis2/cli-helpers-engine').reporter
 const fs = require('fs-extra')
 const path = require('path')
 
-const isNotDotFolder = value => value !== '.' && value !== '..'
+const isDotFolder = name => name[0] === '.'
+
+const isNotDotFolder = name => !isDotFolder(name)
 
 const isDotJSFile = name => new RegExp('[.]js$').test(name)
 
-const extractCodemods = (paths, moduleName) => {
-    const { DHIS2_NODE_MODULES } = paths
-    const codemodsPath = path.join(DHIS2_NODE_MODULES, moduleName, 'codemods')
+const isFolder = path => {
+    if (!fs.existsSync(path)) return false
+    return fs.lstatSync(path).isDirectory()
+}
+
+const hasCodemods = codemodsPath =>
+    !!fs.readdirSync(codemodsPath).filter(curFile => {
+        const curPath = path.join(codemodsPath, curFile)
+
+        if (isFolder(curPath)) return false
+        if (!curFile.match(/\.(j|t)sx?$/)) return false
+
+        return true
+    }).length
+
+const getPathOfDependency = (dependency, cwd) => {
+    const packageJsonResolvePath = `${dependency}/package.json`
+
+    try {
+        const paths = [cwd]
+        const resolvedPath = require.resolve(packageJsonResolvePath, { paths })
+        return path.dirname(resolvedPath)
+    } catch (e) {
+        return ''
```

Comment:
Any reason this is an empty string instead of `null` or `undefined`?

Reply:
Yeah. It allows to avoid null checks. In really doesn't matter in this case, it's just a habit to never change the type of a (return) value as that could introduce unforeseen type bugs in my experience